drivers/rtc/rtc_shell.c

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (c) 2023, Prevas A/S <kim.bondergaard@prevas.dk>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/shell/shell.h>
+#include <zephyr/drivers/rtc.h>
+#include <time.h>
+#include <stdlib.h>
+
+/* Formats accepted when setting date and/or time */
+static const char format_iso8601[] = "%FT%T";
+static const char format_time[] = "%T"; /* hh:mm:ss */
+static const char format_date[] = " %F"; /* yyyy-mm-dd */
+
+#if !defined CONFIG_BOARD_NATIVE_POSIX
+
+static const char *consume_chars(const char *s, char *dest, unsigned int cnt)
+{
+	if (strlen(s) < cnt) {
+		return NULL;
+	}
+
+	memcpy(dest, s, cnt);
+	dest[cnt] = '\0';
+
+	return s + cnt;
+}
+
+static const char *consume_char(const char *s, char ch)
+{
+	if (*s != ch) {
+		return NULL;
+	}
+	return ++s;
+}
+
+static const char *consume_date(const char *s, struct tm *tm_time)
+{
+	char year[4 + 1];
+	char month[2 + 1];
+	char day[2 + 1];
+
+	s = consume_chars(s, year, 4);
+	if (!s) {
+		return NULL;
+	}
+
+	s = consume_char(s, '-');
+	if (!s) {
+		return NULL;
+	}
+
+	s = consume_chars(s, month, 2);
+	if (!s) {
+		return NULL;
+	}
+
+	s = consume_char(s, '-');
+	if (!s) {
+		return NULL;
+	}
+
+	s = consume_chars(s, day, 2);
+	if (!s) {
+		return NULL;
+	}
+
+	tm_time->tm_year = atoi(year) - 1900;
+	tm_time->tm_mon = atoi(month) - 1;
+	tm_time->tm_mday = atoi(day);
+
+	return s;
+}
+
+static const char *consume_time(const char *s, struct tm *tm_time)
+{
+	char hour[2 + 1];
+	char minute[2 + 1];
+	char second[2 + 1];
+
+	s = consume_chars(s, hour, 2);
+	if (!s) {
+		return NULL;
+	}
+
+	s = consume_char(s, ':');
+	if (!s) {
+		return NULL;
+	}
+
+	s = consume_chars(s, minute, 2);
+	if (!s) {
+		return NULL;
+	}
+
+	s = consume_char(s, ':');
+	if (!s) {
+		return NULL;
+	}
+
+	s = consume_chars(s, second, 2);
+	if (!s) {
+		return NULL;
+	}
+
+	tm_time->tm_hour = atoi(hour);
+	tm_time->tm_min = atoi(minute);
+	tm_time->tm_sec = atoi(second);
+
+	return s;
+}
+
+static char *strptime(const char *s, const char *format, struct tm *tm_time)
+{
+	/* Reduced implementation of strptime -
+	 * accepting only the 3 different format strings
+	 */
+	if (!strcmp(format, format_iso8601)) {
+		s = consume_date(s, tm_time);
+		if (!s) {
+			return NULL;
+		}
+
+		s = consume_char(s, 'T');
+		if (!s) {
+			return NULL;
+		}
+
+		s = consume_time(s, tm_time);
+		if (!s) {
+			return NULL;
+		}
+
+		return (char *)s;
+
+	} else if (!strcmp(format, format_time)) {
+		return (char *)consume_time(s, tm_time);
+
+	} else if (!strcmp(format, format_date)) {
+		return (char *)consume_date(s, tm_time);
+
+	} else {
+		return NULL;
+	}
+}
+
+#endif
+
+static int cmd_set(const struct shell *sh, size_t argc, char **argv)
+{
+	const struct device *dev = DEVICE_DT_GET(DT_ALIAS(rtc));
```

Comment:
The RTC shell commands should be able to work on any RTC device in the system, not jus the one defined via an alias. Check the other shell modules for reference on how to let the user specify the device.

Reply:
Fixed. It becomes a bit more troublesome to call the commands though. I guess it is very uncommon to have more than one RTC device. Would be nice if the alias could be used.

For my stm board it will now be 

`rtc get rtc@44007800
`

I'll have to run 'device list' first to find the right device name

Reply:
Given that more than one RTC is quite uncommon, I would be OK with using the alias for now, keeping it simple, until we get a better solution for iterating through RTCs, like the one that exists for sensors, or maybe the new `zephyr,rtc` which may become relevant if this PR #57800 is merged :)